runtime/lua/vim/shared.lua

```diff
@@ -425,6 +425,7 @@ function vim.deep_equal(a, b)
   return false
 end
 
+--- @deprecated
```

Comment:
I'd slap a vim.deprecate() on it

Reply:
We can add that once 0.10 is released 😉 :
- 0.10: soft deprecation (docs only)
- 0.11:  hard deprecation (warning)
- 0.12: removed